_episodes/08-edge-detection.md

```diff
@@ -49,430 +45,368 @@ black and and one white:
+The first several lines of the main program are easily recognizable
+at this point: saving the command-line argument, reading the image in
+grayscale, and creating a window.
 
 ~~~
 '''
- * Main program begins here. 
+ * Python script to demonstrate Canny edge detection
+ * with trackbars to adjust the thresholds.
+ *
+ * usage: python CannyTrack.py <filename>
 '''
-# read command-line filename argument
-filename = sys.argv[1]
-
-# load original image as grayscale
-image = cv2.imread(filename = filename, flags = cv2.IMREAD_GRAYSCALE)
-
-# set up display window with trackbars for minimum and maximum threshold
-# values
-cv2.namedWindow(winname = "edges", flags = cv2.WINDOW_NORMAL)
-~~~
-{: .python}
+import skimage
+import skimage.feature
+import skimage.viewer
+import sys
 
-Then, the program creates two variables to
-hold first guesses for the low and high threshold values, `minT` and `maxT`. 
 
-~~~
-minT = 30
-maxT = 150
+filename = sys.argv[1]
+image = skimage.io.imread(fname=filename, as_gray=True)
+viewer = skimage.viewer.ImageViewer(image)
 ~~~
 {: .python}
 
-Next comes the code where we attach two trackbars to the display window named
-"edges".
+The `skimage.viewer.plugins.Plugin` class is designed in order to manipulate images.
+It takes an `image_filter` argument that should be a function.
+This callable should produce a new image as an output which then will be automatically displayed in the image viewer.
+With this in mind, we write a function to perform Canny filtering, with an image as the first parameter, followed by sigma, and low and high threshold values
 
 ~~~
-cv2.createTrackbar("minT", "edges", minT, 255, adjustMinT)
-cv2.createTrackbar("maxT", "edges", maxT, 255, adjustMaxT)
+def canny(image, sigma, low_threshold, high_threshold):
+    return skimage.feature.canny(
+        image=image,
+        sigma=sigma,
+        low_threshold=low_threshold,
+        high_threshold=high_threshold)
 ~~~
 {: .python}
 
-The `cv2.createTrackbar()` function takes five parameters. Unfortunately, the
-function does not support named parameters. First is a string 
-containing the label that will be used for the trackbar when it is displayed. 
-Next is a string containing the name of the window the trackbar should be 
-attached to. Third is the initial value for the trackbar. Fourth is the maximum
-value for the trackbar; the minimum is always 0. Finally, we pass in the name 
-of a function that will be called whenever the value of the trackbar is changed
-by the user. Here we pass in `adjustMinT` for the minimum threshold trackbar 
-and `adjustMaxT` for the maximum threshold trackbar. 
+Next we create a plugin with our `canny()` as a filter function and add sliders to manipulate function parameters interactively.
+Note that the filter function will be called with the slider parameters according to their names as keyword arguments.
+So make sure to name the sliders appropriately.
```

Comment:
"So make sure . . ." is a fragment and should be part of the preceding sentence.